### ISSUE TITLE
a 'final' NamedTuple is flagged with pyright 1.1.401

### ISSUE THREAD

**Issue by tpgillam** [microsoft/pyright#10490]
Consider:

```python
import typing

@typing.final
class Moo(typing.NamedTuple):
    x: int
    y: float
```

```console
$ uv run --with pyright==1.1.400 pyright moo.py
WARNING: there is a new pyright version available (v1.1.400 -> v1.1.401).
Please install the new version or set PYRIGHT_PYTHON_FORCE_VERSION to `latest`

0 errors, 0 warnings, 0 informations

$ uv run --with pyright==1.1.401 pyright moo.py
/SNIP/.cache/uv/archive-v0/zIVt7Dtwp850HkvvxOiHh/lib/python3.12/site-packages/pyright/dist/dist/typeshed-fallback/stdlib/_typeshed/_type_checker_internals.pyi
  /SNIP/.cache/uv/archive-v0/zIVt7Dtwp850HkvvxOiHh/lib/python3.12/site-packages/pyright/dist/dist/typeshed-fallback/stdlib/_typeshed/_type_checker_internals.pyi:52:91 - error: Unnecessary "# type: ignore" comment (reportUnnecessaryTypeIgnoreComment)
/SNIP/moo.py
  /SNIP/moo.py:5:7 - error: Variables defined in abstract base class are not initialized in final class "Moo"
    Instance variable "_field_defaults" is defined in abstract base class "NamedTuple" but not initialized
    Instance variable "_fields" is defined in abstract base class "NamedTuple" but not initialized
    Instance variable "__orig_bases__" is defined in abstract base class "NamedTuple" but not initialized (reportUninitializedInstanceVariable)
2 errors, 0 warnings, 0 informations
```

(We can see #10487 too, but I'm identifying the second error here).

Note that this error only appears when the type is marked with `typing.final`.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.402.

**Comment by tpgillam** [comment]
Unfortunately I still can reproduce the failure above with pyright 1.1.402

```console
$ uv run --with pyright==1.1.402 pyright moo.py
/SNIP/moo.py
  /SNIP/moo.py:4:7 - error: Variables defined in abstract base class are not initialized in final class "Moo"
    Instance variable "_field_defaults" is defined in abstract base class "NamedTuple" but not initialized
    Instance variable "_fields" is defined in abstract base class "NamedTuple" but not initialized
    Instance variable "__orig_bases__" is defined in abstract base class "NamedTuple" but not initialized (reportUninitializedInstanceVariable)
1 error, 0 warnings, 0 informations
```

**Comment by erictraut** [comment]
OK, I'll reopen and investigate further. I wasn't able to repro the problem you reported. I assumed it was the same issue as https://github.com/microsoft/pyright/issues/10487 since the symptoms were similar.

I'm still not able to repro the problem as reported. I create a source file called "moo.py" and paste in the code from above, then I run `uv run --with pyright==1.1.402 pyright moo.py`. The result is ` errors, 0 warnings, 0 informations`. Do you have any other repro steps?

**Comment by tpgillam** [comment]
Ah apologies - `reportUninitializedInstanceVariable` is disabled by default (even in 'strict' mode), so it must be enabled explicitly.

**Comment by erictraut** [comment]
Ah yes, I'm able to repro the issue now. Thanks.

**Comment by Rupt** [comment]
I have pyright pinned to `1.1.400` in my work to avoid this issue, but I see the errors because my VSCode pylance extension uses its own (upgraded) version.

**Comment by tpgillam** [comment]
Confirmed that this is fixed in 1.1.404 - thank you!
